### PR TITLE
fusionauth-jwt support

### PR DESCRIPTION
**Pull request by skjolber**
Something along there lines? Have not had time time to even run it once yet.

### CODE REVIEW

File: frameworks/fusionauth-jwt-bench/src/main/java/com/github/skjolber/bench/fusionauth/FusionAuthJsonWebTokenVerifier.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+		if(!audience.equals(jwt.getString("aud"))) {
+			throw new InvalidJWTException("Unexpected audience");
+		}
+
+		return jwt;
+	}
+
+	public JWT parseToken(String token) throws Exception {
+		return decoder.decode(token, nullVerifier);
```
Comment (robotdan) [inline frameworks/fusionauth-jwt-bench/src/main/java/com/github/skjolber/bench/fusionauth/FusionAuthJsonWebTokenVerifier.java]: If this is only intended to marshal the string token into a JWT w/out validation, you could use a utility method here so you don't have to pass in a no-op verifier.

```
public JWT parseToken(String token) throws Exception {
  return JWTUtils.decodePayload(token);
}
```

File: frameworks/fusionauth-jwt-bench/src/main/java/com/github/skjolber/bench/fusionauth/FusionAuthJsonWebTokenVerifier.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	private final String audience;
+
+	public FusionAuthJsonWebTokenVerifier(KeyPair keyPair, String issuer, String audience) {
+		this.issuer = issuer;
+		this.audience = audience;
+		
+		String encode = PEM.encode(keyPair.getPublic());
+		
+		decoder = JWT.getDecoder();
+		verifier = RSAVerifier.newVerifier(encode);
```
Comment (robotdan) [inline frameworks/fusionauth-jwt-bench/src/main/java/com/github/skjolber/bench/fusionauth/FusionAuthJsonWebTokenVerifier.java]: There is now an `RSAVerifier.newVerifier` method that takes a public key object. 

For example:

```
verifier = RSAVerifier.newVerifier((RSAPublicKey) keyPair.getPublic());
```

File: frameworks/fusionauth-jwt-bench/src/main/java/com/github/skjolber/bench/fusionauth/FusionAuthJsonWebTokenVerifier.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	private final String issuer;
+	private final String audience;
+
+	public FusionAuthJsonWebTokenVerifier(KeyPair keyPair, String issuer, String audience) {
+		this.issuer = issuer;
+		this.audience = audience;
+		
+		String encode = PEM.encode(keyPair.getPublic());
+		
+		decoder = JWT.getDecoder();
```
Comment (robotdan) [inline frameworks/fusionauth-jwt-bench/src/main/java/com/github/skjolber/bench/fusionauth/FusionAuthJsonWebTokenVerifier.java]: This is fine, but is not necessary for performance. The usage in `verifyJsonWebToken` could be `JWT.getDecoder().decode(token, verifier);` But what you have is just fine if that is your preference.

Comment (skjolber) [reply frameworks/fusionauth-jwt-bench/src/main/java/com/github/skjolber/bench/fusionauth/FusionAuthJsonWebTokenVerifier.java]: I prefer it this way, as it creates one less object.

File: frameworks/fusionauth-jwt-bench/src/main/java/com/github/skjolber/bench/fusionauth/FusionAuthJsonWebTokenVerifier.java (hunk 1 above)
Comment (skjolber) [reply frameworks/fusionauth-jwt-bench/src/main/java/com/github/skjolber/bench/fusionauth/FusionAuthJsonWebTokenVerifier.java]: Yeah thats better and more specific to getting the (body) contents.

**Pull request merged**